src/Lib.P2P/SwarmService.cs

```diff
@@ -0,0 +1,1132 @@
+        }
+
+#pragma warning disable VSTHRD100 // Avoid async void methods
+        /// <summary>
+        ///   Called when a remote peer is connecting to the local peer.
+        /// </summary>
+        /// <param name="stream">
+        ///   The stream to the remote peer.
+        /// </param>
+        /// <param name="local">
+        ///   The local peer's address.
+        /// </param>
+        /// <param name="remote">
+        ///   The remote peer's address.
+        /// </param>
+        /// <remarks>
+        ///   Establishes the protocols of the connection.  Any exception is simply
+        ///   logged as warning.
+        /// </remarks>
+        private async void OnRemoteConnect(Stream stream, MultiAddress local, MultiAddress remote)
+#pragma warning restore VSTHRD100 // Avoid async void methods
+        {
+            if (!IsRunning)
+            {
+                try
+                {
+                    stream.Dispose();
+                }
+                catch (Exception)
+                {
+                    // eat it.
+                }
+
+                return;
+            }
+
+            // If the remote is already trying to establish a connection, then we
+            // can just refuse this one.
+            if (!_pendingRemoteConnections.TryAdd(remote, null))
+            {
+                _log.Debug($"Duplicate remote connection from {remote}");
+                try
+                {
+                    stream.Dispose();
+                }
+                catch (Exception)
+                {
+                    // eat it.
+                }
+
+                return;
+            }
+
+            try
+            {
+                if (_log.IsDebugEnabled)
+                {
+                    _log.Debug($"remote connect from {remote}");
+                }
+
+                // TODO: Check the policies
+
+                var connection = new PeerConnection
+                {
+                    IsIncoming = true,
+                    LocalPeer = LocalPeer,
+                    LocalAddress = local,
+                    LocalPeerKey = LocalPeerKey,
+                    RemoteAddress = remote,
+                    Stream = stream
+                };
+
+                // Are we communicating to a private network?
+                if (NetworkProtector != null)
+                {
+                    connection.Stream = await NetworkProtector.ProtectAsync(connection).ConfigureAwait(false);
+                }
+
+                // Mount the protocols.
+                MountProtocols(connection);
+
+                // Start the handshake
+                // TODO: Isn't connection cancel token required.
+                _ = connection.ReadMessagesAsync(default);
+
+                // Wait for security to be established.
+                await connection.SecurityEstablished.Task.ConfigureAwait(false);
+
+                // TODO: Maybe connection.LocalPeerKey = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)